osh/sh_expr_eval.py

```diff
@@ -730,7 +734,17 @@ def Eval(self, node):
                             array_val = cast(value.BashArray, UP_left)
                             small_i = mops.BigTruncate(
                                 self.EvalToBigInt(node.right))
-                            s = word_eval.GetArrayItem(array_val.strs, small_i)
+                            s, error_code = word_eval.GetArrayItem(
+                                array_val.strs, small_i)
+                            if error_code == 1:
+                                # Note: Bash outputs warning but does not make
+                                # it a real error.  We follow the Bash behavior
+                                # here.
+                                self.errfmt.Print_(
```

Comment:
Another thing we can do is make a hard error when `shopt --set strict_array`, which we already have

There is an `e_strict()` pattern that we sometimes use for that

Though that increases the test matrix, so maybe we can do that in another pass

Reply:
Moved to #2168.